packages/common/src/hooks/useSavedCollections.ts

```diff
@@ -1,84 +1,92 @@
-import { useCallback, useEffect, useState } from 'react'
+import { useCallback, useMemo } from 'react'
 
 import { useDispatch, useSelector } from 'react-redux'
 
+import { ID } from 'models/Identifiers'
 import { Status } from 'models/Status'
+import { CommonState } from 'store/index'
+import { getFetchedCollectionIds } from 'store/saved-collections/selectors'
 
-import { accountActions } from '../store/account'
 import {
+  CollectionType,
   savedCollectionsActions,
   savedCollectionsSelectors
 } from '../store/saved-collections'
 
-const { fetchSavedPlaylists } = accountActions
 const { fetchCollections } = savedCollectionsActions
 
-const {
-  getAccountAlbums,
-  getSavedAlbumsState,
-  getFetchedAlbumsWithDetails,
-  getAccountPlaylists
-} = savedCollectionsSelectors
+const { getAccountAlbums, getSavedCollectionsState, getAccountPlaylists } =
+  savedCollectionsSelectors
 
 const DEFAULT_PAGE_SIZE = 50
 
 export function useSavedAlbums() {
   return useSelector(getAccountAlbums)
 }
 
-/* TODO: Handle filtering
- * Option 1: This hook takes the list of album ids to fetch and computes the unfetched
- * based on that.
- * Option 2: Bake filter into selectors which drive this. Downside: Can't use this in multiple places...
- */
-type UseSavedAlbumDetailsConfig = {
+export function useSavedPlaylists() {
+  return useSelector(getAccountPlaylists)
+}
+
+type UseFetchedCollectionsConfig = {
+  collectionIds: ID[]
+  type: CollectionType
   pageSize?: number
 }
-export function useSavedAlbumsDetails({
+
+type UseFetchedSavedCollectionsResult = {
+  /** A list of IDs representing the subset of requested collections which have been fetched */
+  data: ID[]
+  /** The current fetching state of the list of collections requested */
+  status: Status
+  /** Whether any items remain unfetched */
+  hasMore: boolean
+  /** Triggers fetching of the next page of items */
+  fetchMore: () => void
+}
+/** Given a list of collectionIds and a type ('albums' or 'playlists'), returns state
+ * necessary to display a list of fully-fetched collections of that type, as well as
+ * load any remaining items which haven't been fetched.
+ */
+export function useFetchedSavedCollections({
```

Comment:
nitty nit: curious for your thought process in naming this `useFetchedSavedCollections` rather than `useFetchSavedCollections`?

Reply:
Naming is the hardest problem in software engineering :-)

I had originally just named this `useSavedCollections`, but changed it based on some good feedback to indicate the difference between the minimal saved collection records we read from the `account`, and the full records that we have to fetch from the `/playlists` endpoint. In my mind, the idea here is that it's retrieving the list of collections which have had their full records fetched. That could be through this hook, or elsewhere, they only have to be in the cache. The selector used under the hood will map the "minimal" records across all cached collections and return the intersection.

Also I think I'm just being picky about grammar? 😅  `useFetched...` reads cleaner to me than `useFetch` as the primary value being returned is the list of collections, not the `fetch` function?